references/depth/stereo/train.py

```diff
@@ -0,0 +1,750 @@
+        logger.add_meter(meter_name, fmt="{global_avg:.4f}")
+    if "f1" not in args.metrics:
+        logger.add_meter("f1", fmt="{global_avg:.4f}")
+        
+    if can_do_distributed_test:
+        should_forward = True
+        module = model
+    else:
+        should_forward = args.rank == 0
+        # doing forward on rank 0 will trigger a synchronization
+        # which will not be fulfilled by non rank 0 processes
+        module = model.module
+
+    num_processed_samples = 0
+    with torch.cuda.amp.autocast(enabled=args.mixed_precision, dtype=torch.float16):
+        if should_forward:
+            for blob in metric_logger.log_every(val_loader, print_freq, header):
+                image_left, image_right, disp_gt, valid_disp_mask = (x.to(device) for x in blob)
+                padder = utils.InputPadder(image_left.shape, mode=padder_mode)
+                image_left, image_right = padder.pad(image_left, image_right)
+
+                disp_predictions = module(image_left, image_right, flow_init=None, num_iters=iterations)
+                disp_pred = disp_predictions[-1][:, :1, :, :]
+                disp_pred = padder.unpad(disp_pred)
+
+                metrics, _ = utils.compute_metrics(disp_pred, disp_gt, valid_disp_mask, metrics=logger.meters.keys())
+                num_processed_samples += image_left.shape[0]
+                for name in metrics:
+                    logger.meters[name].update(metrics[name], n=1)
+    
+    num_processed_samples = utils.reduce_across_processes(num_processed_samples)
+    
+    print("Num_processed_samples: ", num_processed_samples)
+    if num_processed_samples != len(val_dataset):
+        warnings.warn(
+            f"Number of processed samples {num_processed_samples} is different"
+            f"from the dataset size {len(val_dataset)}. This may happen if"
+            "the dataset is not divisible by the batch size. Try lowering the batch size for more accurate results."
+        )
+
+    if writter is not None and args.rank == 0:
+        for meter_name, meter_value in logger.meters.items():
+            scalar_name = f"{meter_name} {header}"
+            writter.add_scalar(scalar_name, meter_value.avg, step)
+    
+    logger.synchronize_between_processes()
+    print(header, logger)
+
+
+def evaluate(model, args, writter=None, step=None):
+    val_datasets = args.test_datasets or []
+
+    if args.weights and args.test_only:
+        weights = torchvision.models.get_weight(args.weights)
+        trans = weights.transforms()
+
+        def preprocessing(image_left, image_right, disp, valid_disp_mask):
+            image_left, image_right = trans(image_left, image_right)
+            if disp is not None and not isinstance(disp, torch.Tensor):
+                disp = torch.from_numpy(disp)
+            if valid_disp_mask is not None and not isinstance(valid_disp_mask, torch.Tensor):
+                valid_disp_mask = torch.from_numpy(valid_disp_mask)
```

Comment:
I think we should handle the case where the image is resized, i.e. need to resize disp and valid_disp_mask as well.

Reply:
In that case we would have to move into the `_presets.py` transform to accept optional disparity masks just in case. It still feels odd for me to have presets that are destined to be input only to handle that. 

My suggestion would  be to compute the image size / shape before and after and just scale it inside this function.

Reply:
Although I agree that this may not be the best approach, as of now my recommendation is to follow the train script in segmentation: https://github.com/pytorch/vision/blob/main/references/segmentation/train.py#L42

Basically manually transform the mask and disparity after knowing the image size.